PCbuild/generate_pyconfig.ps1

```diff
@@ -0,0 +1,50 @@
+#
+# Generates pyconfig.h from PC\pyconfig.h.in
+#
+
+param (
+    [string[]]$define,
+    [string]$File
+)
+
+$definedValues = @{}
+
+foreach ($arg in $define) {
+    $parts = $arg -split '='
+
+    if ($parts.Count -eq 1) {
+        $key = $parts[0]
+        $definedValues[$key] = ""
+    } elseif ($parts.Count -eq 2) {
+        $key = $parts[0]
+        $value = $parts[1]
+        $definedValues[$key] = $value
+    } else {
+        Write-Host "Invalid argument: $arg"
+        exit 1
+    }
+}
+
+$cpythonRoot = Split-Path $PSScriptRoot -Parent
+$pyconfigPath = Join-Path $cpythonRoot "PC\pyconfig.h.in"
+
+$header = "/* pyconfig.h.  Generated from PC\pyconfig.h.in by generate_pyconfig.ps1.  */"
+
+$lines = Get-Content -Path $pyconfigPath
+$lines = @($header) + $lines
+
+foreach ($i in 0..($lines.Length - 1)) {
+    if ($lines[$i] -match "^#undef (\w+)$") {
+        $key = $Matches[1]
+        if ($definedValues.ContainsKey($key)) {
+            $value = $definedValues[$key]
+            $lines[$i] = "#define $key $value".TrimEnd()
+        } else {
+            $lines[$i] = "/* #undef $key */"
+        }
+    }
+}
```

Comment:
```suggestion
$lines = $lines | %{
    if ($_ -match "^#undef (\w+)$") {
        $key = $Matches[1];
        if ($definedValues.ContainsKey($key)) {
            "#define $key ${definedValues[$key]}".TrimEnd()
        } else {
            "/* #undef $key */"
        }
    } else {
        $_
    }
}
```

Reply:
Okay, that was a bit of a self-indulgent rewrite that didn't actually make it shorter, simpler, or probably faster 😆 

But if you want, we already assume that a copy of Python is available for bootstrapping on Windows (and I believe we require 3.10), so you could make this a Python script. IIRC, `$(HostPython)` has the location - it should be used elsewhere in `regen.targets` if you need examples.